wincode/src/schema/mod.rs

```diff
@@ -2058,6 +2123,92 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_struct_zero_copy_refs() {
+        // Owned zero-copy type.
+        #[derive(SchemaWrite, SchemaRead, Debug, PartialEq, Eq, proptest_derive::Arbitrary)]
+        #[wincode(internal)]
+        #[repr(C)]
+        struct Zc {
+            a: u8,
+            b: [u8; 64],
+        }
+
+        // `Zc`, mirrored with references.
+        #[derive(SchemaWrite, SchemaRead, Debug, PartialEq, Eq)]
+        #[wincode(internal)]
+        #[repr(C)]
+        struct ZcRefs<'a> {
+            a: &'a u8,
+            b: &'a [u8; 64],
+        }
+
+        // `Zc`, wrapped in a reference.
+        #[derive(SchemaWrite, SchemaRead, Debug, PartialEq, Eq)]
+        #[wincode(internal)]
+        #[repr(transparent)]
+        struct ZcWrapper<'a> {
+            data: &'a Zc,
+        }
+
+        impl<'a> From<&'a ZcRefs<'a>> for Zc {
+            fn from(value: &'a ZcRefs<'a>) -> Self {
+                Self {
+                    a: *value.a,
+                    b: *value.b,
+                }
+            }
+        }
+
+        proptest!(proptest_cfg(), |(data in any::<Zc>())| {
+            let serialized = serialize(&data).unwrap();
+            let deserialized = Zc::deserialize(&serialized).unwrap();
+            assert_eq!(data, deserialized);
+
+            let serialized_ref = serialize(&ZcRefs { a: &data.a, b: &data.b }).unwrap();
+            assert_eq!(serialized_ref, serialized);
+            let deserialized_ref = ZcRefs::deserialize(&serialized_ref).unwrap();
+            assert_eq!(data, (&deserialized_ref).into());
+
+            let serialized_wrapper = serialize(&ZcWrapper { data: &data }).unwrap();
+            assert_eq!(serialized_wrapper, serialized);
+            let deserialized_wrapper = ZcWrapper::deserialize(&serialized_wrapper).unwrap();
+            assert_eq!(data, *deserialized_wrapper.data);
+        });
+    }
+
+    #[test]
+    fn test_pod_zero_copy() {
+        #[derive(Debug, PartialEq, Eq, proptest_derive::Arbitrary, Clone, Copy)]
+        #[repr(transparent)]
+        struct Address([u8; 64]);
+
+        #[derive(SchemaWrite, SchemaRead, Debug, PartialEq, Eq, proptest_derive::Arbitrary)]
+        #[wincode(internal)]
+        #[repr(C)]
+        struct MyStruct {
+            #[wincode(with = "Pod<_>")]
+            address: Address,
+        }
+
+        #[derive(SchemaWrite, SchemaRead, Debug, PartialEq, Eq)]
+        #[wincode(internal)]
+        struct MyStructRef<'a> {
+            inner: &'a MyStruct,
```

Comment:
are we able to test here that Pod<_> implements `ZeroCopy`? I guess we would need a field that uses ref to a Pod... maybe
```
#[wincode(with = "&Pod<_>")]
address: &Address,
```
?